n_queen_problem/NQueenProblem.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Implementation of popular N-Queen Problem using Backtracking algorithm.
+ * The size of a chess board is given, suppose n, then we have to suggest such a configuration than n queens could be placed on the chess board and no queen is under attack
+ * There can be a number of possible solutions for a specific board.
+ * This implementation prints only one valid configuration, it can be extended to print all possible valid configurations.
+ * A good example of recursion.
+ * You can see the solution for various sizes by changing the value of Size in 13th line..
+ */
+
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+using namespace std;
+                                        /* ********************************* */
+#define Size 8                          /* Size of the board = (Size X Size) */
+                                        /* ********************************* */
+
+/*
+ * The Queen is safe, if
+ * There's no other Queen in the same row.
+ * There's no other Queen in the same column.
+ * There's no other Queen in the same diagonal.
+ */
+
+bool Queen_Is_Safe(int board[Size][Size], int row, int col)
+{
+    int i,j;
+    for (i = 0; i < col; i++)
+        if (board[row][i]==1)
+            return false;  // * return false, if there's another Queen present in the same row.
+
+    for (i=row, j=col; i>=0 && j>=0; i--, j--)
+        if (board[i][j]==1)
+            return false;  // * return false, if there's another Queen present in the upper diagonal.
+
+    for (i=row, j=col; j>=0 && i<Size; i++, j--)
+        if (board[i][j]==1)
+            return false;  // * return false, if there's another Queen present in the lower diagonal.
+
+    return true;
+}
+
+bool  Find_solution(int board[Size][Size], int col)
+{
+    if (col >= Size)
+        return true;       // * return true,
+
+    for (int i = 0; i < Size; i++)
+    {
+        if ( Queen_Is_Safe(board, i, col) )
+        {
+            board[i][col] = 1;                       // * A queen is placed on (i, col).
+            if (Find_solution(board, col + 1) )      // * Calling Find_solution() to place the rest of the queens.
+                return true;
+            else
+                board[i][col] = 0;                   // * Backtrack
+        }
+    }
+
+    return false;
+}
+
+int main()
+{
```

Comment:
bracket to be added after main() as : `int main() {`
Same thing to be applied for all loops and functions